### DISCUSSION TITLE
create multiple glb from same file

### DISCUSSION THREAD

**Discussion by rossmarinaro** [enable3d/enable3d#258] [Help / Questions]
Not sure if this has been brought up yet, but I'm wondering how I can instantiate multiple "sprites" or instances of a glb file loaded. I tried to make a class, extending a new ExtendedObject3D, only it doesn't create a new model. What am I doing wrong?

Using the Phaser extension:

     class Baddie extends ExtendedObject3D{
      constructor(scene, gltf)
        {
         super();
         this.add(gltf);
         this.position.set(Phaser.Math.Between(0, 14), Phaser.Math.Between(0, 14), Phaser.Math.Between(0, 14))
         scene.third.add.existing(this) ;
        }
      }
  this.third.load.gltf('assets/meshes/meatbaddie.glb').then(gltf => {
    this.spawn = ()=> new Baddie(this, gltf.scene)
    this.time.addEvent({delay: 3000, callback: ()=>this.spawn(), callbackScope:this, repeat: -1})
  });

**Comment by yandeu** [comment]
Something like this:

```js
class Baddie extends ExtendedObject3D {
  constructor(scene, gltf, x, z) {
    super()
    this.add(gltf.scene.children[0])
    this.position.x = x
    this.position.z = z
    scene.add(this)
  }
}

class MainScene extends Scene3D {
  async create() {
    this.warpSpeed()
    this.camera.position.set(2, 2, 4)

    // load only once
    await this.load.preload('box_man', '/assets/glb/box_man.glb')

    const spawn = (x, z) => {
      this.load.gltf('box_man').then(gltf => {
        new Baddie(this.scene, gltf, x, z)
      })
    }

    spawn(0, 1)
    spawn(2, 3)
    spawn(5, 5)
    spawn(-5, 2)
  }
}
```

**Comment by rossmarinaro** [comment]
This didn't quite work out of the box, (needed into include "this.third.load.preload" for the file path glb loader). I got multiple instances to render, however if the model contained multiple objects, I would need to just include the entire "scene" property of the glb, rather than an index of the children array. I noticed some objects would not render this way, but for some reason just calling "gltf.scene" worked..
   example: 

this works in a class extending ExtendedObject3D
      this.add(gltf.scene);

but this misses parts of a model containing multiple objects
    for (let i in gltf.scene.children)
      this.add(gltf.scene.children[i]);

if gltf is just one object, this works fine
      this.add(gltf.scene.children[0]);


Also, if I destroy one instance of my new 3d "sprite", they all disappear. Same goes for removing physics. How can this be prevented?
